### PR TITLE
Add the FifoQueue parameter to sqs.Queues

### PR DESCRIPTION
**Pull request by Lowercases**
Add the FifoQueue parameter from CloudFormation:
http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues.html#cfn-sqs-queue-fifoqueue
as a variable to sqs.Queque.

Please note that this parameter requires Replacement on update.

### CODE REVIEW

**Comment by russellballestrini** [comment]
Will merging this trigger the Replacement or is that something end users can schedule?

**Comment by Lowercases** [comment]
@russellballestrini It won't, however it seems there's a missing piece yet. AWS demands that the name of a FifoQueue end with ".fifo", but troposphere+stacker are unable to create a stack with a dot on it.

**Comment by phobologic** [comment]
@Lowercases Do you have an example that doesn't work?  Honestly, I think we should probably just move this blueprint over to using TroposphereType - I think it should "just work".

The issue you're seeing with the `.fifo` name thing is because it doesn't look like this blueprint uses `QueueName` attribute of the SQS Queue resource.  You're setting the Cloudformation resource name to use `.fifo` most likely, and Cloudformation resources can't have periods in them. Make sense?

**Comment by Lowercases** [comment]
@phobologic the latest versions of Troposphere work indeed with these queues.

Added now a check for QueueName to end with `.fifo` in case FifoQueue is provided, since CloudFormation will enforce it.

**Comment by phobologic** [comment]
@Lowercases cool - honestly though, I think we should pull the `.fifo` thing in troposphere directly, since this would be more generally useful.  Want to pull that out of here and submit a PR to troposphere?

**Comment by phobologic** [comment]
You can see an example of how you'd add this kind of a validation here:

https://github.com/cloudtools/troposphere/blob/779ba85f7e1577e16161a26da480c9b892541471/troposphere/codebuild.py#L20

**Comment by Lowercases** [comment]
@mike LMK what you think! https://github.com/cloudtools/troposphere/compare/master...Lowercases:validate-fifo-queues

File: stacker_blueprints/sqs.py [hunk 1]
```diff
@@ -25,6 +26,14 @@ def validate_queue(queue):
     if "RedrivePolicy" in queue:
         queue["RedrivePolicy"] = sqs.RedrivePolicy(**queue["RedrivePolicy"])
 
+    if "FifoQueue" in queue and queue["FifoQueue"]:
```
Comment (phobologic) [inline stacker_blueprints/sqs.py]: I think what you want here is:

```
if queue.get("FifoQueue"):
```

`.get` on a dict returns None if the key doesn't exist.

File: stacker_blueprints/sqs.py [hunk 2]
```diff
@@ -25,6 +26,14 @@ def validate_queue(queue):
...
+        if "QueueName" not in queue or \
```
Comment (phobologic) [inline stacker_blueprints/sqs.py]: Similarly, this could be replaced with:
```
if not queue.get("QueueName", "").endswith(".fifo"):
```

**Comment by phobologic** [comment]
Also - you really should add this `.fifo` logic to troposphere - we'd love to add it :)

**Comment by Lowercases** [comment]
Submitted with suggestions to https://github.com/cloudtools/troposphere/pull/757

**Pull request merged**